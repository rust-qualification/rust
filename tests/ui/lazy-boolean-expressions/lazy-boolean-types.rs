// Lazy Boolean Opearations: 8ee2a7bd-59c9-4479-9516-5519af52aa3b
// Lazy Boolean Opearations: 44418fad-561e-488e-9a04-30a4701aa735

use std::any::type_name;
use std::any::type_name_of_val;

pub fn main() {
    let result: bool = true && true;
    assert_eq!(type_name::<bool>(), type_name_of_val(&result));

    assert!(true && false);
    assert!(true || false);
    assert!(1 && 1); 
    //~^ ERROR mismatched types
    //~| ERROR mismatched types
    assert!(1.5 && 1.5); 
    //~^ ERROR mismatched types
    //~| ERROR mismatched types
    assert!(1.5 && 1); 
    //~^ ERROR mismatched types
    //~| ERROR mismatched types    
    assert!(false && 1); 
    //~^ ERROR mismatched types
    assert!('a' && 'b'); 
    //~^ ERROR mismatched types
    //~| ERROR mismatched types    
    assert!('a' || true); 
    //~^ ERROR mismatched types
    assert!("alpha" && "beta");
    //~^ ERROR mismatched types
    //~| ERROR mismatched types     
    assert!("alpha" || true); 
    //~^ ERROR mismatched types
}