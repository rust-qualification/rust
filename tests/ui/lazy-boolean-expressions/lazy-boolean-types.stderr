error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:13:13
   |
LL |     assert!(1 && 1); 
   |             ^ expected `bool`, found integer

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:13:18
   |
LL |     assert!(1 && 1); 
   |                  ^ expected `bool`, found integer

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:14:13
   |
LL |     assert!(1.5 && 1.5); 
   |             ^^^ expected `bool`, found floating-point number

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:14:20
   |
LL |     assert!(1.5 && 1.5); 
   |                    ^^^ expected `bool`, found floating-point number

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:15:13
   |
LL |     assert!(1.5 && 1); 
   |             ^^^ expected `bool`, found floating-point number

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:15:20
   |
LL |     assert!(1.5 && 1); 
   |                    ^ expected `bool`, found integer

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:16:22
   |
LL |     assert!(false && 1); 
   |             -----    ^ expected `bool`, found integer
   |             |
   |             expected because this is `bool`

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:17:13
   |
LL |     assert!('a' && 'b'); 
   |             ^^^ expected `bool`, found `char`

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:17:20
   |
LL |     assert!('a' && 'b'); 
   |                    ^^^ expected `bool`, found `char`

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:18:13
   |
LL |     assert!('a' || true); 
   |             ^^^ expected `bool`, found `char`

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:19:13
   |
LL |     assert!("alpha" && "beta"); 
   |             ^^^^^^^ expected `bool`, found `&str`

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:19:24
   |
LL |     assert!("alpha" && "beta"); 
   |                        ^^^^^^ expected `bool`, found `&str`

error[E0308]: mismatched types
  --> $DIR/lazy-boolean-types.rs:20:13
   |
LL |     assert!("alpha" || true); 
   |             ^^^^^^^ expected `bool`, found `&str`

error: aborting due to 13 previous errors

For more information about this error, try `rustc --explain E0308`.
